lib/confirm-controller.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+var util = require('util');
+var Controller = require('./base-controller');
+var _ = require('lodash');
+
+var ConfirmController = function ConfirmController(options) {
+  Controller.apply(this, arguments);
+  this.options = options || {};
+};
+
+util.inherits(ConfirmController, Controller);
+
+/*
+ * Utility function which returns the
+ * name of the step a field is present on
+ */
+function getStepFromFieldName(fieldName, steps) {
+  return _.findKey(steps, function findKey(step) {
+    if (!step.fields) {
+      return false;
+    }
+    return step.fields.indexOf(fieldName) > -1;
+  });
+}
+
+/*
+ * Utility function to create an array
+ * of field objects made up of field name,
+ * value and step
+ */
+function getFieldObjectsFromNames(fields, values, modifiers, steps) {
```

Comment:
Can these arguments be key:values in an `params` object?
Using an object will alleviate some of the pain of needing to check for their presence, and simplifies the interface, i.e., which arguments are mandatory?